process_report/institute_list_validate.py

```diff
@@ -0,0 +1,47 @@
+import sys
+import argparse
+import pydantic
+import yaml
+
+from process_report.institute_list_models import InstituteList
+
+
+def pydantic_to_github(err, rates_file):
+    """Produce a github error annotation from a pydantic ValidationError"""
+    for error in err.errors():
+        print(f"::error file={rates_file},title=Validation error::{error['msg']}")
+
+
+def yaml_to_github(err, rates_file):
+    """Produce a github error annotation from a YAML ParserError"""
+    line = err.problem_mark.line
+    print(
+        f"::error file={rates_file},line={line},title=Parser error::{err.context}: {err.problem}"
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-g", "--github", action="store_true", help="Emit github workflow annotations"
+    )
+    parser.add_argument(
+        "institute_list", default="process_report/institute_list.yaml", nargs="?"
+    )
+    args = parser.parse_args()
+
+    try:
+        with open(args.institute_list) as f:
+            data = yaml.safe_load(f)
+            institute_list = InstituteList.model_validate(data)
+            print(
+                f"Validation of {len(institute_list.root)} institution entries successful"
+            )
+    except pydantic.ValidationError as err:
+        if args.github:
+            pydantic_to_github(err, args.rates_file)
```

Comment:
There is no argument named `rates_file`.

Reply:
I've added a simple test case for the validator